### PR TITLE
The controllers recognize and apply input with a deadband of 0.2 pres…

### PR DESCRIPTION
**Pull request by JacobGH111**
…ent.

### CODE REVIEW

File: competition/src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -75,11 +76,14 @@ public void teleopInit() {
...
     // Get the x speed. We are inverting this because Xbox controllers return
     // negative values when we push forward.
-    double xSpeed = -m_xspeedLimiter.calculate(m_controller.getLeftY()) * Drivetrain.kMaxSpeed;
+    double deadband = 0.2;
+    double yControllerInput = MathUtil.applyDeadband(m_controller.getLeftY(), deadband);
+    double xControllerInput = MathUtil.applyDeadband(m_controller.getLeftX(), deadband);
+    double xSpeed = -m_xspeedLimiter.calculate(yControllerInput) * Drivetrain.kMaxSpeed;
     if(Math.abs(m_controller.getLeftY()) < 0.1) {
       xSpeed = 0;
     }
```
Comment (amorygalili) [inline competition/src/main/java/frc/robot/Robot.java:85]: Remove old deadband

File: competition/src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -75,11 +76,14 @@ public void teleopInit() {
...
-    double ySpeed = m_yspeedLimiter.calculate(m_controller.getLeftX()) * Drivetrain.kMaxSpeed;
+    double ySpeed = m_yspeedLimiter.calculate(xControllerInput) * Drivetrain.kMaxSpeed;
     if(Math.abs(m_controller.getLeftX()) < 0.1) {
       ySpeed = 0;
     }
```
Comment (amorygalili) [inline competition/src/main/java/frc/robot/Robot.java:89]: Remove old deadband

File: competition/src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (JacobGH111) [reply competition/src/main/java/frc/robot/Robot.java:85]: your a dry commenter

File: competition/src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (JacobGH111) [reply competition/src/main/java/frc/robot/Robot.java:89]: spice it up

File: competition/src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -80,13 +80,6 @@ public void teleopPeriodic() {
     double yControllerInput = MathUtil.applyDeadband(m_controller.getLeftY(), deadband);
     double xControllerInput = MathUtil.applyDeadband(m_controller.getLeftX(), deadband);
     double xSpeed = -m_xspeedLimiter.calculate(yControllerInput) * Drivetrain.kMaxSpeed;
-    if(Math.abs(m_controller.getLeftY()) < 0.1) {
-      xSpeed = 0;
-    }
-    double ySpeed = m_yspeedLimiter.calculate(xControllerInput) * Drivetrain.kMaxSpeed;
```
Comment (amorygalili) [inline competition/src/main/java/frc/robot/Robot.java]: You need this line

**Review by amorygalili — CHANGES_REQUESTED** [review]

File: competition/src/main/java/frc/robot/Robot.java (hunk 3 above)
Comment (JacobGH111) [reply competition/src/main/java/frc/robot/Robot.java]: =^.^=

# So Boring
(the comment)

**Review by amorygalili — APPROVED** [review]

**Pull request merged**